tools/pipelines-tasks/AVDAppAttachPublish/main.ts

```diff
@@ -0,0 +1,136 @@
+import tl = require('azure-pipelines-task-lib/task');
+import { ToolRunner } from 'azure-pipelines-task-lib/toolrunner';
+import path = require('path');
+import armStorage = require('azure-pipelines-tasks-azure-arm-rest/azure-arm-storage');
+import { AzureRMEndpoint } from 'azure-pipelines-tasks-azure-arm-rest/azure-arm-endpoint';
+import { AzureEndpoint, StorageAccount } from 'azure-pipelines-tasks-azure-arm-rest/azureModels';
+import * as telemetry from "azure-pipelines-tasks-utility-common/telemetry";
+import * as os from 'os';
+import * as url from 'url';
+import * as fs from 'fs';
+import helpers = require('common-helpers/helpers');
+
+const HELPER_SCRIPT = path.join(__dirname, 'publishAVD.ps1');
+const TARGET_DLL = path.join(__dirname, '/AppAttachFrameworkDLL/AppAttachKernel.dll');
+const AppAttachFrameworkDll = path.join(__dirname, '/AppAttachFrameworkDLL');
+
+const HELPER_SCRIPT_EXCEPTIONS_TELEMETRY = path.join(__dirname, 'ReportExceptionTelemetry.ps1');
+const TARGET_DLL_TELEMETRY = path.join(__dirname, '/AppAttachFrameworkDLL/AppAttachTelemetry.dll');
+
+function isNonEmpty(str: string): boolean {
+    return (!!str && !!str.trim());
+}
+
+function getResourceGroupNameFromUri(resourceUri: string): string {
+	if (isNonEmpty(resourceUri)) {
+		const parsedUrl = url.parse(resourceUri, true);
+
+		const pathname = parsedUrl.pathname || '';
+		const segments = pathname.split('/');
+
+		const resourceGroupIndex = segments.indexOf('resourceGroups');
+
+		if (resourceGroupIndex !== -1 && resourceGroupIndex < segments.length - 1) {
+			return segments[resourceGroupIndex + 1];
+		}
+	}
+
+    return "";
+}
+
+async function run(): Promise<void> {
+	let isSuccessful = true;
+	try {
+		await helpers.installNuget(helpers.APPATTACH_FRAMEWORK_NUPKG_DIR, AppAttachFrameworkDll, 'net472');
+
+		let connectedServiceName = tl.getInput('connectedServiceNameARM', true)!;
+
+		let subscriptionID: string = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true)!;
+
+		let storageAccountName = tl.getInput('storageAccount', true)!;
+		var azureEndpoint: AzureEndpoint = await new AzureRMEndpoint(connectedServiceName).getEndpoint();
+		let accessToken: string = (await azureEndpoint.applicationTokenCredentials.getToken()).toString();
```

Comment:
Can we authenticate user before env preparation?

Reply:
user authentication happens when service connection is established. and these storage account details are fetched only when it is successful.